### ISSUE TITLE
No syslog messages if server is offline when app starts

### ISSUE THREAD

**Issue by otherdave** [luigiberrettini/NLog.Targets.Syslog#110]
I'm not sure which details to provide, but I have a syslog server running on System A and my app is running on System B.  I'm using UDP to log.

If System A is in the middle of coming online when my app starts on System B, I don't see any syslog messages. If I restart my application or the entire System B, I see messages as expected.

Is this expected or intentional behavior? Is there a way around it?

**Comment by otherdave** [comment]
My nlog target config for my app. I borrowed most of it from you sample/test app (thanks for that!). The syslog server is always running at 172.31.1.1.

```xml
  <target xsi:type="sl:Syslog" name="syslogger">
            <sl:layout xsi:type="SimpleLayout" text="${message}${exception:format=ToString,StackTrace}" />
            <sl:enforcement>
                <sl:splitOnNewLine>true</sl:splitOnNewLine>
                <sl:transliterate>true</sl:transliterate>
                <sl:replaceInvalidCharacters>true</sl:replaceInvalidCharacters>
                <sl:truncateFieldsToMaxLength>true</sl:truncateFieldsToMaxLength>
                <sl:truncateMessageTo>1024</sl:truncateMessageTo>
            </sl:enforcement>
            <sl:messageCreation>
                <sl:facility>Local4</sl:facility>
                <sl:rfc>Rfc5424</sl:rfc>
            </sl:messageCreation>
            <sl:messageSend>
                <sl:protocol>UDP</sl:protocol>
                <sl:udp>
                    <sl:server>172.31.1.1</sl:server>
                    <sl:port>514</sl:port>
                </sl:udp>
                <sl:tcp>
                    <sl:server>172.31.1.1</sl:server>
                    <sl:port>514</sl:port>
                    <sl:reconnectInterval>3</sl:reconnectInterval>
                    <sl:framing>octetCounting</sl:framing>
                </sl:tcp>
            </sl:messageSend>
        </target>
```

**Comment by luigiberrettini** [comment]
Message transmission involves:
 - [SyslogTarget line 54](https://github.com/graffen/NLog.Targets.Syslog/blob/ea78af0bd489a92e0e2d426ac39724a03be2f74a/src/NLog.Targets.Syslog/SyslogTarget.cs#L54)
 - [AsyncLogger line 36](https://github.com/graffen/NLog.Targets.Syslog/blob/ea78af0bd489a92e0e2d426ac39724a03be2f74a/src/NLog.Targets.Syslog/AsyncLogger.cs#L36)
 - [AsyncLogger line 76](https://github.com/graffen/NLog.Targets.Syslog/blob/ea78af0bd489a92e0e2d426ac39724a03be2f74a/src/NLog.Targets.Syslog/AsyncLogger.cs#L76)
 - [LogEventMsgSet line 58](https://github.com/graffen/NLog.Targets.Syslog/blob/master/src/NLog.Targets.Syslog/LogEventMsgSet.cs#L58)
 - [MessageTransmitter line 28](https://github.com/graffen/NLog.Targets.Syslog/blob/master/src/NLog.Targets.Syslog/MessageSend/MessageTransmitter.cs#L28)
 - [Udp](https://github.com/graffen/NLog.Targets.Syslog/blob/master/src/NLog.Targets.Syslog/MessageSend/Udp.cs) and [Tcp](https://github.com/graffen/NLog.Targets.Syslog/blob/master/src/NLog.Targets.Syslog/MessageSend/Tcp.cs)

When the target is initialized both an UdpClient and a TcpClient are created.
Since UDP is connectionless this should not impact sends as happens for TCP where a proper reconnect is implemented.

If I remember well the test application works when you send messages, then start the fake syslog server and then send new messages.

I suggest using the test application against your server to check if the problem is elsewhere.
Another temporary solution could be reconfiguring the target programmatically when you see the server is up (INotifyPropertyChanged is implemented and reinitializes the target).

**Comment by otherdave** [comment]
@luigiberrettini Thanks for the replies! Sorry to go silent for a bit, I've had to step away from this for a few days. I should be able to run the test you suggested today or tomorrow and I'll report back.

**Comment by otherdave** [comment]
Ok, looks like I'm the problem :)

Ran my server and stopped the rsyslog service. Then I launched your GUI test app (after configuring it to point to my system) and sent off some logs. Verified nothing was logged. Started rsyslog and sent a few more logs from your GUI and they showed up.

Either my previous test was flawed, or I've got another error somewhere. I'll dig into it and report back if something else is wacky. Thanks for the help & advice!

Does your GUI & nlog config buffer UDP syslogs, or otherwise hang onto them? I didn't look too closely at the config but I noticed a lag of many seconds between clicking the "Info Event" button and the log message making it to my rsyslog log file.
